### PR TITLE
Upgrade k8petstore example to v1beta3 API.

### PR DESCRIPTION
**Pull request by pedro-r-marques**
Convert the example to the v1beta3 API since support for beta1 was removed.

### CODE REVIEW

**Comment by pedro-r-marques** [comment]
I signed the CLA

**Comment by pedro-r-marques** [comment]
Added my gmail address (which signed the CLA) to my github account. Googlebot, please verify my CLA acceptance.

**Comment by bgrant0607** [comment]
cc @nikhiljindal @caesarxuchao @krousey

File: examples/k8petstore/k8petstore.sh [hunk 1]
```diff
@@ -46,173 +46,193 @@ function create {
...
       }
     }
-  },
-  "labels": {"name": "redisslave"}
+  }
 }
 EOF
-$kubectl create -f rm.json --api-version=v1beta1 --namespace=$NS
-$kubectl create -f rm-s.json --api-version=v1beta1 --namespace=$NS
+$kubectl create -f rm.json --api-version=v1beta3 --namespace=$NS
```
Comment (bgrant0607) [inline examples/k8petstore/k8petstore.sh]: FYI, you don't need to specify the api version here, since it's specified in the json.

**Comment by bgrant0607** [comment]
Hey, Pedro, thanks for the update.

We do have a CLA, signed today. Not sure what's up with the bot.

File: examples/k8petstore/k8petstore.sh [hunk 2]
```diff
@@ -46,173 +46,193 @@ function create {
...
+         "containers": [{
+           "name": "frontend-go-restapi",
+           "image": "jayunit100/k8-petstore-web-server:$VERSION"
          }]
-         }
        },
-       "labels": {
-         "name": "frontend",
-         "uses": "redis-master"
+       "metadata": {
```
Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: Should there be  a "name": "frontendCcontroller" after this line?

Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: A nit, could you keep the order to "spec" and "metadata" consistent?

File: examples/k8petstore/k8petstore.sh [hunk 3]
```diff
@@ -46,173 +46,193 @@ function create {
...
+    "labels": {
+      "name": "frontend"
+    }
   },
-  "labels": {
-    "name": "frontend"
+  "spec": {
+    "ports": [{
+      "port": 3000,
+      "targetPort": 3000
```
Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: The default value of "targetPort" is the same as "port", https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/api/v1beta3/types.go#L1108

Please omit the default value.

File: examples/k8petstore/k8petstore.sh [hunk 4]
```diff
@@ -46,173 +46,193 @@ function create {
...
+    "labels": {
+      "name": "redis-master"
+    }
   },
-  "labels": {
-    "name": "redis-master"
+  "spec": {
+    "ports": [{
+      "port": 6379,
+      "targetPort": 6379
```
Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: Please omit the default value.

File: examples/k8petstore/k8petstore.sh [hunk 5]
```diff
@@ -46,173 +46,193 @@ function create {
...
+    "labels": {
+      "name": "redisslave"
+    }
   },
-  "selector": {
-    "name": "redisslave"
+  "spec": {
+    "ports": [{
+      "port": 6379,
+      "targetPort": 6379
```
Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: Same here.

**Comment by caesarxuchao** [comment]
@pedro-r-marques, just a few nits, otherwise LGTM. Thank you for the contribution!

File: examples/k8petstore/k8petstore.sh (hunk 2 above)
Comment (bgrant0607) [inline examples/k8petstore/k8petstore.sh]: This is the pod template. It doesn't need a name.

File: examples/k8petstore/k8petstore.sh (hunk 3 above)
Comment (bgrant0607) [inline examples/k8petstore/k8petstore.sh]: This and other best practices are described in #7257.

File: examples/k8petstore/k8petstore.sh (hunk 2 above)
Comment (caesarxuchao) [inline examples/k8petstore/k8petstore.sh]: @pedro-r-marques, please remove the name fields of other pod templates as well, thanks.

**Comment by caesarxuchao** [comment]
LTGM. Thank you @pedro-r-marques !

**Pull request merged**